.azure/pipelines/jobs/default-build.yml

```
@@ -249,7 +249,7 @@ jobs:
         condition: always()
 
     # Run component detection after all successful Build:* jobs unless overridden e.g. for Alpine build.
-    - ${{ if and(startsWith(parameters.jobDisplayName, 'Build:'), ne(variables['skipComponentGovernanceDetection'], 'true'), ne(parameters.skipComponentGovernanceDetection, 'true'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+    - ${{ if and(startsWith(parameters.jobDisplayName, 'Build:'), eq(parameters.agentOs, 'NotARealOperatingSystem'), ne(variables['skipComponentGovernanceDetection'], 'true'), ne(parameters.skipComponentGovernanceDetection, 'true'), notin(variables['Build.Reason'], 'PullRequest')) }}:
```

Suggest including a reason e.g.
```suggestion
    # Disabled for now because detector limits some error handling to the auto-injected task. Will re-enable once fixed.
    - ${{ if and(startsWith(parameters.jobDisplayName, 'Build:'), eq(parameters.agentOs, 'NotARealOperatingSystem'), ne(variables['skipComponentGovernanceDetection'], 'true'), ne(parameters.skipComponentGovernanceDetection, 'true'), notin(variables['Build.Reason'], 'PullRequest')) }}:
```

---

Might want to mention microsoft/componentdetection-azdo#181 for tracking too

---

```suggestion
    # See microsoft/componentdetection-azdo#181
    - ${{ if and(startsWith(parameters.jobDisplayName, 'Build:'), eq(parameters.agentOs, 'NotARealOperatingSystem'), ne(variables['skipComponentGovernanceDetection'], 'true'), ne(parameters.skipComponentGovernanceDetection, 'true'), notin(variables['Build.Reason'], 'PullRequest')) }}:
```

---

I filed an issue https://github.com/dotnet/aspnetcore/issues/30126 and noted it as part of the BuildOps log to track.